### ISSUE TITLE
esm typings are weak or I have done something wrong

### ISSUE THREAD

**Issue by kirevdokimov** [toji/gl-matrix#427]
# TLDR

Types from `npm run build-dts` and esm build from `npm run build-esm` don't work together 

---

Plain project, no bundlers, just Typescript 3.9.6 with "es6" target to compile.

I was going to use gl-matrix as a bundled dependency because of no bundler. I got esm dist from `npm run build-esm` then copy to my `./src/esm` dir. Works just fine, but no types (like `mat4.create()` returns `any` instead of `mat4` so I can't handle it with Typescript)

![image](https://user-images.githubusercontent.com/5202070/116794883-42c89980-aad9-11eb-8179-d2303a72a17d.png)

So I have made `index.d.ts` from `npm run build-dts`. Keeping this in root folder provides no types, for this reason I just copied this right to `./src/esm`. But now I got
```
src/Cube.ts:7:22 - error TS2306: File '/mnt/c/Project/subtract-n-triangulate/src/esm/index.d.ts' is not a module.

7 import { mat4 } from "./esm/index.js";
                       ~~~~~~~~~~~~~~~~
Found 1 error.
```

For sure I can not an expert in typescript module declarations, but something looks wrong with all this. Would be happy to get some advice or solution. Thank you.

**Comment by kirevdokimov** [comment]
Ok, got it.

First of all we declare our wrap module but never export it from `index.d.ts`:
https://github.com/toji/gl-matrix/blob/fccd5f707432ec5cc1774575be84c9f901a198d0/utils/bundle-dts.js#L37-L38

![image](https://user-images.githubusercontent.com/5202070/116796013-dbfbae00-aae1-11eb-824e-5f4c60bc19e2.png)

---

https://github.com/toji/gl-matrix/blob/fccd5f707432ec5cc1774575be84c9f901a198d0/utils/bundle-dts.js#L10
<img src="https://user-images.githubusercontent.com/5202070/116796115-b1f6bb80-aae2-11eb-9415-e6c295106c4e.png" width="200">

---

Second, even if we replace `declare` with `export` we still don't have types, because of `module "gl-matrix"` wrap. There is probably a reason for this to be here, but if I remove it, everything works just as I initially expect.
